pkg/handlers/webhook.go

```diff
@@ -148,15 +148,15 @@ func (h *OptlyWebhookHandler) HandleWebhook(w http.ResponseWriter, r *http.Reque
 
 	// Iterate through all SDK keys and update config
 	for _, sdkKey := range webhookConfig.SDKKeys {
-		h.optlyCache.UpdateConfigs(sdkKey)
-
 		if h.syncEnabled {
 			if err := h.configSyncer.Sync(r.Context(), syncer.GetDatafileSyncChannel(), sdkKey); err != nil {
 				errMsg := fmt.Sprintf("datafile synced failed. reason: %s", err.Error())
 				log.Error().Msg(errMsg)
+				h.optlyCache.UpdateConfigs(sdkKey)
```

Comment:
For the webhook receiver node, isn't this redundant to `updateConfig` in line 187? We can skip this one here.